### PR TITLE
Ngwmn 1835 site pages fetching wrong data

### PR DESCRIPTION
**Pull request by abriggs-usgs**
Before making a pull request
----------------------------

- [x] Make sure all tests run
- [x] Update the changelog appropriately

NGWMN 1835 Site Pages Fetching Wrong Data
-----------

Description
-----------
For monitoring locations with the same geographical coordinates, a query to GeoServer will return a set of locations in a JSON format. Code was added to sort through the query results for the desired monitoring location based on the site id.

After making a pull request
---------------------------
- [ ] If appropriate, put the link to the PR in the JIRA ticket
- [ ] Assign someone to review unless the change is trivial

### CODE REVIEW

File: server/ngwmn/tests/test_views.py [hunk 1]
```diff
@@ -49,7 +60,22 @@ def test_best_case(self, mocker):
...
         mocker.get(self.stats_monthly_url, text=self.mock_monthly_json, status_code=200)
 
-        response = self.app_client.get(self.site_loc_url)
+        response = self.app_client.get(self.site_loc_url_1)
+        self.assertEqual(response.status_code, 200)
+
+    # Tests if a nested site (a monitoring location with the same geographic coordinates as another monitoring location)
+    # can be selected by site id
+    @requests_mock.Mocker()
+    def test_nested_monitoring_location(self, mocker):
```
Comment (mbucknell) [inline server/ngwmn/tests/test_views.py]: I don't think this second test really tests if the correct feature is returned (The above test would have worked fine before your change). One way to check that the correct feature is used would be check the contents of the response for <title> tag for the site name. You would want distinct SITE_NAME property for each feature then so would need to update the TEST_SUMMARY_JSON. You could check the same thing on the first test to make sure it's the other SITE_NAME.

File: server/ngwmn/views.py [hunk 2]
```diff
@@ -69,7 +70,7 @@ def provider(agency_cd):
                            data_collection=pull_feed(confluence_url(agency_cd, DATA_COLLECTION_CONTENT)),
                            data_management=pull_feed(confluence_url(agency_cd, DATA_MANAGEMENT_CONTENT)),
                            other_agency_info=pull_feed(confluence_url(agency_cd, OTHER_AGENCY_INFO_CONTENT))
-                          )
+                           )
```
Comment (mbucknell) [inline server/ngwmn/views.py]: Codacy is complaining about indentation here.

File: server/ngwmn/tests/test_views.py (hunk 1 above)
Comment (abriggs-usgs) [reply server/ngwmn/tests/test_views.py]: Okay. I think I made the changes needed to confirm that the correct monitoring location is in the response.

File: server/ngwmn/views.py [hunk 3]
```diff
@@ -105,7 +106,15 @@ def site_page(agency_cd, location_id):
     )
     water_quality = get_water_quality(agency_cd, location_id)
 
-    feature = summary['features'][0]['properties']
+    # reduce the amount of data returned from GeoServer to only the monitoring location of interest
+    # this section is important for monitoring locations that have the same geographical coordinates (i.e. nested wells)
+    feature = {}
+    index_of_feature = 0
+    while index_of_feature < len(summary['features']):
```
[the remark this answers is not in the document]
Comment (abriggs-usgs) [reply server/ngwmn/views.py]: I cleaned this up a bit. Not sure if could be improved more.

File: server/ngwmn/tests/test_views.py [hunk 4]
```diff
@@ -24,23 +23,37 @@ class TestWellPageView(TestCase):
...
+        self.stats_overall_url_2 = '/'.join([SERVICE_ROOT, _stats_url, 'wl-overall', _agency_cd, _location_id_2])
+        self.stats_monthly_url_2 = '/'.join([SERVICE_ROOT, _stats_url, 'wl-monthly', _agency_cd, _location_id_2])
+
         self.mock_site_info_json = json.dumps(MOCK_SITE_INFO)
         self.mock_overall_json = json.dumps(MOCK_OVERALL_STATS)
         self.mock_monthly_json = json.dumps(MOCK_MONTHLY_STATS)
 
     @requests_mock.Mocker()
     def test_best_case(self, mocker):
+        id_of_expected_nested_location = b'430406089232901'
```
Comment (mbucknell) [inline server/ngwmn/tests/test_views.py]: This variables are confusingly named. How about calling it id1 and id2, set the values in the setup, and adjust the tests accordingly. Our use case for this issue happens to be "nested" sites. Naming them as such here just confuses the issue.

Comment (abriggs-usgs) [reply server/ngwmn/tests/test_views.py]: Makes sense to me. Will do.

**Pull request merged**